nsxt/resource_nsxt_policy_nat_rule.go

```diff
@@ -345,8 +362,8 @@ func resourceNsxtPolicyNATRuleCreate(d *schema.ResourceData, m interface{}) erro
 		ruleStruct.TranslatedPorts = &ports
 	}
 
-	if len(sNets) > 0 {
-		ruleStruct.SourceNetwork = &sNets
+	if sNets != nil && len(*sNets) > 0 {
```

Comment:
I think since sNets would be nil unless specified, we can move it up into struct initialization, no need for this if (same below)

Reply:
Done. Also added a test case to test create NO_SNAT with no translated network